privacy/activitycontrol.go

```diff
@@ -88,34 +57,17 @@ func activityRulesToEnforcementRules(rules []config.ActivityRule) ([]Rule, error
 			result = ActivityDeny
```

Comment:
I noticed that this line doesn't have any code coverage. It may be out of the scope of this PR's goals, so I'm fine leaving it as is, but wanted to point it out.

Reply:
Yeah, we only tests it with allow=true. 

Here is a quick fix.
These functions should take a parameter that will control the result:

```go

func getTestActivityConfig(allow bool) config.Activity {
	return config.Activity{
		Default: ptrutil.ToPtr(true),
		Rules: []config.ActivityRule{
			{
				Allow: allow,
				Condition: config.ActivityCondition{
					ComponentName: []string{"bidderA"},
					ComponentType: []string{"bidder"},
				},
			},
		},
	}
}

func getTestActivityPlan(activityRes ActivityResult) ActivityPlan {
	return ActivityPlan{
		defaultResult: true,
		rules: []Rule{
			ConditionRule{
				result:        activityRes,
				componentName: []string{"bidderA"},
				componentType: []string{"bidder"},
			},
		},
	}
}
```
Put "true" and "ActivityAllow" to everywhere where these parameters are needed. 
And then modify existing test to receive false at least for one activity:

```go
{
			name: "specified_and_correct",
			privacyConf: config.AccountPrivacy{
				AllowActivities: &config.AllowActivities{
					SyncUser:                 getTestActivityConfig(true),
					FetchBids:                getTestActivityConfig(true),
					EnrichUserFPD:            getTestActivityConfig(true),
					ReportAnalytics:          getTestActivityConfig(true),
					TransmitUserFPD:          getTestActivityConfig(true),
					TransmitPreciseGeo:       getTestActivityConfig(false),
					TransmitUniqueRequestIds: getTestActivityConfig(true),
					TransmitTids:             getTestActivityConfig(true),
				},
			},
			activityControl: ActivityControl{plans: map[Activity]ActivityPlan{
				ActivitySyncUser:                 getTestActivityPlan(ActivityAllow),
				ActivityFetchBids:                getTestActivityPlan(ActivityAllow),
				ActivityEnrichUserFPD:            getTestActivityPlan(ActivityAllow),
				ActivityReportAnalytics:          getTestActivityPlan(ActivityAllow),
				ActivityTransmitUserFPD:          getTestActivityPlan(ActivityAllow),
				ActivityTransmitPreciseGeo:       getTestActivityPlan(ActivityDeny),
				ActivityTransmitUniqueRequestIds: getTestActivityPlan(ActivityAllow),
				ActivityTransmitTids:             getTestActivityPlan(ActivityAllow),
			}},
		},
```

Reply:
Good catch @AlexBVolcy. Thanks for the implementation plan @VeronikaSolovei9. I'll add this to the PR tomorrow.